src/Control/Alternative/Free.hs

```diff
@@ -47,14 +39,8 @@ infixl 3 `Ap`
 data AltF f a where
   Ap     :: f a -> Alt f (a -> b) -> AltF f b
   Pure   :: a                     -> AltF f a
-#if __GLASGOW_HASKELL__ >= 707
-  deriving Typeable
```

Comment:
Are you sure you want to remove the `Typeable` instance? The trivial change would be to keep it unconditionally.

Reply:
(Similarly in all other analogous places)

Reply:
I actually didn't remove it, as GHC automatically defines `Typeable` instances for all type and data constructors from 7.10 onwards. In fact, the `Data` instances defined elsewhere in this library wouldn't typecheck without the auto-defined `Typeable` instances!

Reply:
Ah, I wasn't aware, thanks!